### PR TITLE
[Breaking] Fix .predict() method and add .predict_proba() in xgboost.dask.DaskXGBClassifier

### LINKED ISSUE #5985: xgboost.dask.DaskXGBClassifier implementation of .predict() does not adhere to the sklearn API specification
[Sklearn Specification for Classifiers](https://scikit-learn.org/stable/developers/develop.html#specific-models)

The `.predict()` method of `xgboost.dask.DaskXGBClassifier` currently returns probabilities. Per the specification, the `.predict()` method is supposed to return class labels. This is also inconsistent with the behavior of the `.predict()` method of `xgboost.XGBClassifier`, which properly returns class labels.

Other functionality in `dask` (specifically in `dask_ml.model_selection`) depend on the behavior being correct.

Example of correct behavior in `xgboost.XGBClassifier`:

```
import xgboost as xgb
from sklearn.datasets import make_classification

X, y = make_classification(n_samples=1000, n_informative=5, n_classes=2, random_state=1234)
clf = xgb.XGBClassifier(objective="binary:logistic")
clf.fit(X, y)
print(clf.predict(X)[:5])
# [0 0 1 1 1]
```

Example of incorrect behavior in `xgboost.dask.DaskXGBClassifier`:

```
import xgboost as xgb
import dask.dataframe as dd
import dask.distributed
from sklearn.datasets import make_classification


cluster = dask.distributed.LocalCluster(n_workers=2, threads_per_worker=1)
client = dask.distributed.Client(cluster)
X, y = make_classification(n_samples=1000, n_informative=5, n_classes=2, random_state=1234)
X_ = dd.from_array(X, chunksize=500)
y_ = dd.from_array(y, chunksize=500)
clf = xgb.dask.DaskXGBClassifier(objective="binary:logistic")
clf.fit(X_, y_)
print(clf.predict(X_).compute()[:5])
# [0.03111755 0.00773133 0.99876463 0.99792993 0.9944484 ]
```

### PR DESCRIPTION
**Pull request by jameskrach**
Fixes #5985

This is cannot be merged before #5984 is resolved because [this line](https://github.com/jameskrach/xgboost/blob/fix-dask-sklearn-api/python-package/xgboost/dask.py#L1103) will raise an exception about there not being another axis if the model is a multiclass classifier.

### CODE REVIEW

**Comment by trivialfis** [comment]
Thanks for the PR.  I submitted a fix for #5984 https://github.com/dmlc/xgboost/pull/5989 .

**Comment by hcho3** [comment]
Marking this PR as "breaking," since it changes the behavior of `predict()`.

@trivialfis Should this be part of 1.2.0 release?

**Comment by trivialfis** [comment]
Will look into the details tonight.

File: python-package/xgboost/dask.py [hunk 1]
```diff
@@ -1079,13 +1079,32 @@ def fit(self, X, y,
...
+    def predict_proba(self, data):  # pylint: disable=arguments-differ,missing-docstring
+        _assert_dask_support()
+        return self.client.sync(self._predict_proba_async, data)
+
+    async def _predict_async(self, data):
+        _assert_dask_support()
+
+        test_dmatrix = await DaskDMatrix(client=self.client, data=data,
+                                         missing=self.missing)
+        pred_probs = await predict(client=self.client,
```
Comment (trivialfis) [inline python-package/xgboost/dask.py:1098]: In the next release I want to replace this with inplace predict.

**Comment by jameskrach** [comment]
I'd make the change myself for inplace predict but to be honest I'm not familiar with what it does. Would you like me to add a `#TODO(@trivialfis): ...`? Definitely agree that while it's breaking it's more of a bugfix.

Also, sorry for all of the CI runs - I made a small donation to offset (hopefully all of) the cost.

**Comment by trivialfis** [comment]
No problem, your contribution is very valuable.  I'm really happy that you can participate in the development of dask interface and hoping that I can see you more often in the future!

>  Would you like me to add a #TODO(@trivialfis): ...?

Not necessary as it's on my to do list, avoid making data copy is quite important on our side.

**Comment by jameskrach** [comment]
Ah I missed one `_assert_dask_support()` call. Apologies for the last change.

**Comment by trivialfis** [comment]
> Ah I missed one _assert_dask_support() call. Apologies for the last change.

Thanks and no problem!  But is it already in the `predict_prob` function?

**Comment by jameskrach** [comment]
Yes, but both the private `_predict_async()` and public `predict()` methods had the call before, so I added it for consistency.

**Comment by trivialfis** [comment]
Got it.  I will run it tomorrow (away from my desktop right now).

**Comment by trivialfis** [comment]
@hcho3 please let me handling the merging for this PR.  :-)

**Comment by hcho3** [comment]
@trivialfis Sure. I'll make RC2 after you merge this PR.

**Comment by trivialfis** [comment]
@jameskrach Thanks for the excellent work on predict proba!
